### PR TITLE
fix(ui): DOMA-5408 fixed RadioGroup component

### PR DESCRIPTION
**Pull request by Alllex202**
Need for https://github.com/open-condo-software/condo/pull/2821

## Before
Simple `RadioGroup` does not work.
Only list with `RadioGrop`s
Снимок экрана 2023-02-22 в 17 29 41

## After
Simple `RadioGroup` does work.
Снимок экрана 2023-02-22 в 17 29 55

And list too
Снимок экрана 2023-02-22 в 17 29 41

### CODE REVIEW

**Review by sitozzz — APPROVED** [review]

File: packages/ui/src/components/Radio/radiogroup.tsx [hunk 1]
```diff
@@ -15,16 +16,18 @@ type RadioGroupOptionType = Pick<RadioProps, 'value' | 'label'> & {
...
     options: Array<RadioGroupOptionType>
 }
 
+// TODO(DOMA-5430): breaking changes (v2.0.0)
 export type RadioGroupProps = {
-    icon: React.ReactElement
-    groups: Array<RadioGroupType>
-} & Pick<DefaultRadioGroupProps, 'value' | 'onChange' | 'disabled'>
+    /**
+     * @deprecated
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Radio/radiogroup.tsx:25]: Use `@deprecated <reason>` format

File: packages/ui/src/components/Radio/radiogroup.tsx [hunk 2]
```diff
@@ -57,16 +60,16 @@ const GroupWithIcon: React.FC<RadioGroupType> = ({ name, options, icon }) => {
...
     return (
         <DefaultRadio.Group
             {...rest}
             prefixCls={RADIO_GROUP_CLASS_PREFIX}
         >
-            {groups.map(group => (
-                <GroupWithIcon name={group.name} options={group.options} key={group.name} icon={icon} />
-            ))}
+            {groups ? groups.map(group => (
+                <GroupWithIcon name={group.name} options={group.options} key={group.name} />
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Radio/radiogroup.tsx:71]: GroupWithIcon must be exported as well, since we decided to use children prop

Comment (SavelevMatthew) [reply packages/ui/src/components/Radio/radiogroup.tsx:71]: I guess it should be <Radio.GroupWithIcon> or smth like that....name is up to you

**Review by SavelevMatthew — APPROVED** [review]

**Review by pahaz — APPROVED** [review]

File: packages/ui/src/components/Radio/radiogroup.tsx [hunk 3]
```diff
@@ -15,16 +16,22 @@ type RadioGroupOptionType = Pick<RadioProps, 'value' | 'label'> & {
...
-} & Pick<DefaultRadioGroupProps, 'value' | 'onChange' | 'disabled'>
+    /**
+     * @deprecated should use "children" and "RadioGroup.ItemGroup" instead of "groups". "groups" will be removed in the next major release (v2).
+     */
+    groups?: Array<RadioGroupType>
+} & Pick<DefaultRadioGroupProps, 'value' | 'onChange' | 'disabled' | 'children'>
 
-const GroupWithIcon: React.FC<RadioGroupType> = ({ name, options, icon }) => {
+type CompoundedComponent = React.FC<RadioGroupProps> & {
+    ItemGroup: typeof GroupWithIcon
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Radio/radiogroup.tsx:31]: why not strict type?

File: packages/ui/src/components/Radio/radiogroup.tsx [hunk 4]
```diff
@@ -15,16 +16,22 @@ type RadioGroupOptionType = Pick<RadioProps, 'value' | 'label'> & {
...
+}
+
+const GroupWithIcon: React.FC<RadioGroupType> = ({ name, options }) => {
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Radio/radiogroup.tsx:34]: Maybe we should rename with to `ItemGroup` as well?

File: packages/ui/src/components/Radio/radiogroup.tsx (hunk 3 above)
Comment (Alllex202) [reply packages/ui/src/components/Radio/radiogroup.tsx]: fixed it

File: packages/ui/src/components/Radio/radiogroup.tsx (hunk 4 above)
Comment (Alllex202) [reply packages/ui/src/components/Radio/radiogroup.tsx]: fixed it

**Pull request merged**